js/state/syncUtil.js

```diff
@@ -194,10 +232,22 @@ module.exports.applySyncRecord = (record) => {
  */
 module.exports.applySyncRecords = (records) => {
   if (!records || records.length === 0) { return }
+  const siteRecords = []
+  const otherRecords = []
+  records.forEach((record) => {
```

Comment:
bookmark folders:

instead of applying all bookmarks, i think we should first apply all folderless bookmarks and folders, then apply foldered bookmarks. otherwise we won't be able to resolve bookmark.parentFolderObjectId -> parentFolderId, so bookmarks will end up folderless.

bonus points to recursively apply (top level folders/items, then second level, etc) to better ensure folders ids are available when we need them.